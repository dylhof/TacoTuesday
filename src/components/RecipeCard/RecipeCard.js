import React from 'react';
import uuid from 'uuid/v4'

export const RecipeCard = ({ tacoRecipe }) => {
  const splitRecipe = tacoRecipe.recipe.split('\n');
  const jsxRecipeItems = splitRecipe.map(step => {
    if (!step.includes('=')) {
      return (
        <li key={uuid()}>
          {step}
        </li>
      )
    }
  });
  return (
    <div className='recipeCard'>
      <h3>{tacoRecipe.name}</h3>
      <ul dense={true} className='recipeList'>
        {jsxRecipeItems}
      </ul>
    </div>
  );
};

export default RecipeCard;