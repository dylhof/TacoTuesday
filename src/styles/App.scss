*,
*::before,
*::after {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.App {
  background-image: url('./images/tacoBackground.jpg');
  background-size: cover;
}

