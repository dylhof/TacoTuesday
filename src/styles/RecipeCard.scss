.recipeCard {
  margin: 10px;
  padding: 10px;
  opacity: .8;
  min-width: 300px;
  border-radius: 5px;
  background-color: $highlight;

  .recipeListDiv{
  }
  
  .recipeList {
    margin: 10px;
    overflow: scroll;
    border: 2px solid $secondary;
    border-radius: 5px;
    padding: 3px;
  }
}

.homeRecipes {
  height: 300px;
  
  .homeList {
    height: 250px
  }
}