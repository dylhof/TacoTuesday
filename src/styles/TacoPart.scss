.carouselDisplay {
  display: flex;
  align-content: center;
  align-items: center;
  justify-content: center;
}

.carousel {
  display: flex;
  width: 90%;
  overflow: scroll;
}