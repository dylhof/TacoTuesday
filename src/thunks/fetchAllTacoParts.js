import { fetchCall } from '../utils/Api'
import * as actions from '../actions'

export const fetchAllTacoParts = () => {
  const proxyURL = 'https://cors-anywhere.herokuapp.com/'
  const tacoURL = 'http://taco-randomizer.herokuapp.com/'
  const tacoParts = ['base_layers', 'mixins', 'seasonings', 'condiments', 'shells']
  return async dispatch => {
    try {
      dispatch(actions.setLoading(true))
      tacoParts.forEach(async tacoPart => {
        const actionName = 'set'+ tacoPart[0].toUpperCase() + tacoPart.slice(1)
        const tacoPartRecipes = await fetchCall(`${proxyURL}${tacoURL}${tacoPart}`)
        dispatch(actions[actionName](tacoPartRecipes))
      })
      dispatch(actions.setLoading(false))
    } catch (error) {
      dispatch(actions.setError('ERROR Tacos Not Loading!'))
    }
  }
}