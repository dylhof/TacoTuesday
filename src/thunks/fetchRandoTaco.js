import { fetchCall } from '../utils/Api';
import { setLoading, setRandoTaco, setError} from '../actions'

export const fetchRandoTaco = () => {
  const proxyURL = 'https://cors-anywhere.herokuapp.com/'
  const tacoURL = 'http://taco-randomizer.herokuapp.com/random'
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      const randoTaco = await fetchCall(proxyURL + tacoURL);
      console.log('fromfetchrandoTaco', randoTaco)
      dispatch(setLoading(false));
      dispatch(setRandoTaco(randoTaco));
    } catch (error) {
      dispatch(setError(
        'Taco bout embarassing! We have encountered a taco loading error :('
      ))
    }
  }
}