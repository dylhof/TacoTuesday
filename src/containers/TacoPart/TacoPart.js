import React, { Component } from 'react';
import { connect } from 'react-redux';
import RecipeCard from '../../components/RecipeCard/RecipeCard';
import uuid from 'uuid/v4';

export class TacoPart extends Component {
  shiftTaco = (event) => {
    debugger
    const shiftDirection = event.target.classList.contains('left') ? -1 : 1
    const currentPosition = event.target.parentElement.children[1].scrollLeft;
    event.target.parentElement.childNodes[1].scrollLeft = currentPosition +(600 * shiftDirection)
  }

  render () {
    const part = this.props.tacoPart;
    const tacoPartRecipes = this.props[part].map(recipe => {
      return (
        <RecipeCard 
        key={uuid()} 
        tacoRecipe={recipe} styleName='tacoPartRecipes'
        listStyleName='tacoPartList'/>
      )
    });

    return (
      <div className='carouselDisplay'>
      <i className="fas fa-chevron-left fa-2x left" onClick={this.shiftTaco}></i>
        <div className='carousel'>
          {tacoPartRecipes}
        </div>
        <i className="fas fa-chevron-right fa-2x right" onClick={this.shiftTaco}></i>
      </div>
    )
  };
}

export const mapStateToProps = state => ({
  baseLayers: state.baseLayers,
  mixins: state.mixins,
  condiments: state.condiments,
  seasonings: state.seasonings,
  shells: state.shells
})

export default connect(mapStateToProps)(TacoPart);
