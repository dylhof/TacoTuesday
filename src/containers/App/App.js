import React, { Component } from 'react';
import {Header} from '../../components/Header/Header';
import Home from '../Home/Home';
import { ErrorDisplay } from '../../components/ErrorDisplay/ErrorDisplay';
import { withRouter, Switch, Route } from 'react-router-dom';
import {fetchRandoTaco} from '../../thunks/fetchRandoTaco';
import { connect } from 'react-redux';
import {fetchAllTacoParts} from '../../thunks/fetchAllTacoParts';
import TacoPart from '../TacoPart/TacoPart';
import BuildATaco from '../BuildATaco/BuildATaco';
import RecipeCard from '../../components/RecipeCard/RecipeCard';
// import('../../thunks/fetchAllTacoParts').then(fetchAllTacoParts => {fetchAllTacoParts.fetchAllTacoParts()})


export class App extends Component {
  componentDidMount = () => {
    this.props.fetchRandoTaco()
    this.props.fetchAllTacoParts()
  }

  findTacoPart = ({ match }) => {
    const tacoParts = ['baseLayers', 'mixins', 'condiments', 'seasonings', 'shells']
    const tacoPart = tacoParts.find(part => part === match.params.tacoPart)
    return(
      <TacoPart tacoPart={tacoPart}/>
    )
  }

  findTacoPartRecipe = ({ match }) => {
    const { baseLayers, mixins, condiments, seasonings, shells } = this.props
    const allRecipes = [...baseLayers, ...mixins, ...condiments, ...seasonings, ...shells]
    const tacoPartRecipe = allRecipes.find(recipe => recipe.slug === match.params.tacoPartRecipe)
    return (
      <RecipeCard tacoRecipe={tacoPartRecipe} /> 
    )
  }

  render() {
    return (
      <div className="App">
        <Route path='/' component={Header} />
        <Switch>
        <Route path='/' exact component={Home} />
          <Route path='/explore/:tacoPart' render={this.findTacoPart}/>
          <Route path='/build_a_taco' component={BuildATaco} />
          <Route path='/:tacoPartRecipe' render={this.findTacoPartRecipe}/>
          <Route render={ErrorDisplay}/>
        </Switch>
        {this.props.error && <h2>{this.props.error}</h2>}
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  fetchRandoTaco: () => dispatch(fetchRandoTaco()),
  fetchAllTacoParts: () => dispatch(fetchAllTacoParts())
})

export const mapStateToProps = state => ({
  error: state.error,
  isLoading: state.isLoading,
  baseLayers: state.baseLayers,
  mixins: state.mixins,
  condiments: state.condiments,
  seasonings: state.seasonings,
  shells: state.shells
})


export default withRouter(
  connect(mapStateToProps, 
    mapDispatchToProps
    )(App)
);
