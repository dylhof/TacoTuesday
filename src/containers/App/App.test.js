import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';
import  TacoPart from '../TacoPart/TacoPart';
import RecipeCard from '../../components/RecipeCard/RecipeCard';
import {fetchRandoTaco} from '../../thunks/fetchRandoTaco';
import {fetchAllTacoParts} from '../../thunks/fetchAllTacoParts';
import {mapDispatchToProps} from './App';
import {mapStateToProps} from './App';

jest.mock('../../thunks/fetchRandoTaco')
fetchRandoTaco.mockReturnValue([{name: 'taco'}])
jest.mock('../../thunks/fetchAllTacoParts')
fetchAllTacoParts.mockReturnValue([{name: 'taco'}])


describe('App', () => {

  let wrapper
  let mockBaseLayers
  let mockMixins
  let mockSeasonings
  let mockCondiments
  let mockShells

  beforeEach(() => {
    mockBaseLayers = [{slug: 'carnitas'}]
    mockMixins = [{slug: 'lettuce'}]
    mockSeasonings = [{slug: 'salt'}]
    mockCondiments = [{slug: 'salsa'}]
    mockShells = [{slug: 'bad_ass_tortillas'}]
    wrapper = shallow(
    <App
      fetchRandoTaco={fetchRandoTaco}
      fetchAllTacoParts={fetchAllTacoParts}
      baseLayers={mockBaseLayers}
      mixins={mockMixins}
      condiments={mockCondiments}
      seasonings={mockSeasonings}
      shells={mockShells}
    />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should find a matching taco part', () => {
    //setup
    const expected = <TacoPart tacoPart={'mixins'}/>
    //execution
    const result = wrapper.instance().findTacoPart({ match: { params: { tacoPart: 'mixins' } } })
    //expectation
    expect(result).toEqual(expected)
  });

  it.skip('should find a matching recipe', () => {
    //setup
    const slug = 'bad_ass_tortillas'
    const expected =  (<div onClick={() => this.props.history.goBack()}>
                        <RecipeCard tacoRecipe={{slug}} /> 
                      </div>)
    //execution
    const result = wrapper.instance().findTacoPartRecipe({ match: {params: {tacoPartRecipe: slug } } })
    //expectation
    expect(result).toEqual(expected)
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with fetchRandoTaco', () => {
      //setup
      const mockDispatch = jest.fn()
      const expected = fetchRandoTaco()
      //execution 
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.fetchRandoTaco()
      //expectation
      expect(mockDispatch).toHaveBeenCalledWith(expected)
    });

    it('should call dispatch with fetchAllTacoParts', () => {
      //setup
      const mockDispatch = jest.fn()
      const expected = fetchAllTacoParts()
      //execution
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.fetchAllTacoParts()
      //expectation
      expect(mockDispatch).toHaveBeenCalledWith(expected)
    })
  })

  describe('mapStateToProps', () => {
    it('should return an object with all state as props', () => {
      //setup
      const mockState = {
        error: 'No tacos? sad.', 
        isLoading: true,
        baseLayers: [{name: 'taco!'}],
        mixins: [{name: 'taco!'}],
        condiments: [{name: 'taco!'}],
        seasonings: [{name: 'taco!'}],
        shells: [{name: 'taco!'}],
        mockProperty: 'this is NOT a taco'
      }
      const expected = {
        error: 'No tacos? sad.', 
        isLoading: true,
        baseLayers: [{name: 'taco!'}],
        mixins: [{name: 'taco!'}],
        condiments: [{name: 'taco!'}],
        seasonings: [{name: 'taco!'}],
        shells: [{name: 'taco!'}]
      }
      //execution
      const mappedProps = mapStateToProps(mockState)
      //expectation
      expect(mappedProps).toEqual(expected)
    })
  })
})
