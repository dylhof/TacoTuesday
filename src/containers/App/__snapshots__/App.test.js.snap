// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <Route
    component={[Function]}
    path="/"
  />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      path="/explore/:tacoPart"
      render={[Function]}
    />
    <Route
      component={[Function]}
      path="/build_a_taco"
    />
    <Route
      path="/:tacoPartRecipe"
      render={[Function]}
    />
    <Route
      render={[Function]}
    />
  </Switch>
</div>
`;
